modules/issue_tracker/php/module.class.inc

```diff
@@ -74,11 +74,16 @@ class Module extends \Module
      */
     public function getWidgets(string $type, \User $user, array $options) : array
     {
+        if (!in_array($type, ['usertasks', 'candidate'], true)) {
+            return [];
```

Comment:
Would this be a Bad Request?

Reply:
No, there are valid widget types not implemented by this module (`dashboard`, `candidateinfo`..).

This check is just to avoid calling the factory on things that aren't implemented by the switch statement after pulling is out of the individual `case`s.

Reply:
In the context of this module though, wouldn't `dashboard` be a Bad Request? Even if it's valid in other child classes.

Reply:
No. You are allowed to visit the dashboard page even if you have the `candidate_profile` module installed.

Reply:
I'm not really sure what you have in mind - maybe I don't understand the callstack of how getWidgets works.

Anyway this is a minor point so I'll resolve this

Reply:
`getWidgets` is called on all active modules with a `$type` for any module that wants to get all the widgets of that type. If the module doesn't know anything about that type, it doesn't mean it was a bad call, it means that it has no widgets to return. For instance, the dashboard will call `getWidgets('dashboard', ...)` on all modules. If the issue_tracker gets that call and doesn't have any dashboard widgets to return (either because it doesn't know about the dashboard or doesn't care) that doesn't make it a bad call or a bad request from a client, it makes the proper thing to do to return an empty array of no widgets.

Reply:
I suppose that the module is only a API data provider, so Bad Request is not necessary, maybe we can return empty or 501 (Not Implemented) instead of "Bad Request"

Reply:
There is nothing wrong with the request.